lib/raml-fixers/nullable-fixer.js

```diff
@@ -21,14 +27,40 @@ function customizer (input) {
     return;
   }
 
-  const output = _.cloneDeep(input);
-  if ('type' in output && nullable === true) {
-    output.type += ' | nil';
+  //If we are in this point it's because we are the input it's nullable.
+  var output = {};
+  if ('type' in input) { 
+
+    if (input.type === 'object') {
+      //Objects needs to be cloned fixing their nullable childs first.
+      let obj = _.cloneDeepWith(input, objectCustomizer);
+      deleteNullable(obj);
+      output.anyOf = [];
+      output.anyOf.push(obj);
+      output.anyOf.push({'type':'nil'});
+    } else {
+      //Nullable and not an object.
+      output = _.cloneDeep(input);
+      output.type += ' | nil';
+    }
+  } else {
+    //Nullable without a type.
+    output = _.cloneDeep(input);
   }
   deleteNullable(output);
```

Comment:
Recommend moving deleteNullable into scope on line 44 and 48, otherwise it's not clear that deleteNullable doesn't apply to "object" logic above